Transient Tracers

JanStreffing opened the pull request:
I checkout out https://github.com/FESOM/fesom2/tree/2.1-awiesm-2.1-with-transient-tracers and did a git revert on the critical commit https://github.com/FESOM/fesom2/pull/570/commits/fb6eab8aa3bbdf1c0201359bbc841fdafd8507ee.

The result is, that the MR reduces from 279 changes files to 42 changed files, and from +62,015 −116,851 lines to a much more manageable +8,685 −150. A plenitude of conflicts remain though.

@mbutzin, please have a look if this branch still constitutes a working tracers branch, or if undoing that commit anything critical got lost. Note that no src/ files were changed, but lots of namelists were.

mbutzin commented on config/namelist.config.transit:
This namelist.config.transit is obsolete now and can be deleted.

mbutzin commented on src/gen_modules_config.F90:
OK for transient tracers.

mbutzin commented on config/namelist.config:
OK for transient tracerss

mbutzin commented on src/io_meandata.F90:
OK for transient tracers.

mbutzin commented on src/io_restart.F90:
OK for transient tracers but there seems to be a doubling of wiso code in lines 202-236.

mbutzin commented on src/mod_transit.F90:
This is the essential module for transient tracer simulations.

mbutzin commented on src/oce_ale_tracer.F90:
OK for transient tracers

mbutzin commented on src/oce_setup_step.F90:
OK for transient tracers

mbutzin commented on src/fvom_main.F90:
Transient tracer code is only in lines 38-39, 181-192, and 803-817.

mbutzin commented on config/namelist.io.transit:
namelist.io.transit can be remove but namelist.io has to include 'otracers' for transient tracers.

mbutzin commented:
> I checkout out https://github.com/FESOM/fesom2/tree/2.1-awiesm-2.1-with-transient-tracers and did a git revert on the critical commit [fb6eab8](https://github.com/FESOM/fesom2/commit/fb6eab8aa3bbdf1c0201359bbc841fdafd8507ee).
> 
> The result is, that the MR reduces from 279 changes files to 42 changed files, and from +62,015 −116,851 lines to a much more manageable +8,685 −150. A plenitude of conflicts remain though.
> 
> @mbutzin, please have a look if this branch still constitutes a working tracers branch, or if undoing that commit anything critical got lost. Note that no src/ files were changed, but lots of namelists were.



> I checkout out https://github.com/FESOM/fesom2/tree/2.1-awiesm-2.1-with-transient-tracers and did a git revert on the critical commit [fb6eab8](https://github.com/FESOM/fesom2/commit/fb6eab8aa3bbdf1c0201359bbc841fdafd8507ee).
> 
> The result is, that the MR reduces from 279 changes files to 42 changed files, and from +62,015 −116,851 lines to a much more manageable +8,685 −150. A plenitude of conflicts remain though.
> 
> @mbutzin, please have a look if this branch still constitutes a working tracers branch, or if undoing that commit anything critical got lost. Note that no src/ files were changed, but lots of namelists were.

Checked ... Go ahead, @JanStreffing!

JanStreffing commented:
@mbutzin and I had a look at this and came to the conclusion, that many of the code changes and subsequent MR conflicts result from the older realization of iceberg and wiso. Both have today been mergen in a newer version via: https://github.com/FESOM/fesom2/pull/489

I will try to rebase Martins tracer code on top of the current refactoring branch instead. This would result in a much smaller MR.

JanStreffing commented:
Here is the preliminary result with a simple git patch:
```

git status
On branch refactoring_tracers_butzin
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   config/namelist.config
	modified:   config/namelist.forcing
	modified:   src/gen_modules_clock.F90
	modified:   src/oce_ale_tracer.F90

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	config/namelist.config.rej
	config/namelist.io.transit
	config/namelist.oce.cfc
	config/namelist.oce.rej
	src/gen_model_setup.F90.rej
	src/gen_modules_clock.F90.rej
	src/gen_modules_config.F90.rej
	src/io_meandata.F90.rej
	src/io_restart.F90.rej
	src/mod_transit.F90
	src/oce_ale_tracer.F90.rej
	src/oce_setup_step.F90.rej

no changes added to commit (use "git add" and/or "git commit -a")
```

While that is more reject than clean applications, I can already tell from the low number of files affected, that this seems quite doable. @mbutzin Tomorrow we sit down with you to make this happen. I think we can manage the resulting MR for 2.6.

JanStreffing commented:
superseded by https://github.com/FESOM/fesom2/pull/576
